perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1341,6 +1341,29 @@ public RichMember sponsorMember(PerunSession session, Member sponsored, User spo
 		return membersManagerBl.convertMembersToRichMembers(sess, membersManagerBl.getSponsoredMembers(sess, vo));
 	}
 
+	@Override
+	public List<MemberWithSponsors> getSponsoredMembersAndTheirSponsors(PerunSession sess, Vo vo, List<String> attrNames) throws VoNotExistsException, PrivilegeException, AttributeNotExistsException {
+		Utils.checkPerunSession(sess);
+		Utils.notNull(vo, "vo");
+
+		perunBl.getVosManagerBl().checkVoExists(sess, vo);
+
+		//Authorization
+		if(!AuthzResolver.authorizedInternal(sess, "getSponsoredMembersAndTheirSponsors_Vo_policy", vo)) {
+			throw new PrivilegeException(sess, "getSponsoredMembersAndTheirSponsors");
+		}
+
+		List<AttributeDefinition> attrsDef = new ArrayList<>();
+		for (String attrName : attrNames) {
+			attrsDef.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrName));
+		}
+
+		return membersManagerBl.convertMembersToRichMembersWithAttributes(sess, getSponsoredMembers(sess, vo), attrsDef).stream()
```

Comment:
You should also filter only allowed attributes based on the caller authorization, see ``membersManagerBl.filterOnlyAllowedAttributes(sess, richMembers, null, true);``

Reply:
I added this filter to the method.